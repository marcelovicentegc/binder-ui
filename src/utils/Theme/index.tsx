import baseStyled, { ThemedStyledInterface } from "styled-components";

export const theme = {
  importFontFamily:
    "@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Sans&display=swap');",
  useFontFamily: "font-family: IBM Plex Sans, sans-serif;",
  color: {
    primaryBlack: "#262626",
    black1: "#202020",
    black2: "#2B2B2B",
    white1: "#F7F7F7",
    white2: "#DCDDE0", // TODO: move to grey
    white3: "#F5F5F5",
    gray1: "#E9E9E9",
    gray2: "#C4C4C4",
    gray3: "#999999",
    gray4: "#4C4C4C",
    gray5: "#D9D9D9",
    gray6: "8C8C8C",
    gray7: "#4D4D4D",
    gray8: "#E6E6E6",
    gray9: "#5C5C5C",
    gray10: "#EFF6EA",
    red1: "#D24654",
    red2: "#BB3B46",
    red3: "#A52F38",
    red4: "#831D23",
    red5: "#831D23",
    orange1: "#EBA248",
    orange2: "#E2873C",
    orange3: "#D86C30",
    orange5: "#C43618",
    yellow1: "#F4DA4E",
    yellow2: "#EFC841",
    yellow3: "#EAB634",
    yellow4: "#DA7F0D",
    green1: "#5DC957",
    green2: "#4DAE48",
    green3: "#3E933A",
    green4: "#276B24",
    green5: "#0F420E",
    blue1: "#6ACEDF",
    blue2: "#59B5CF",
    blue3: "#479DBF",
    blue4: "#3584AF",
    blue5: "#12538F",
    purple1: "#5A4D80",
    rainbowColors: [
      "#CC0000",
      "#F06D00",
      "#E9CD00",
      "#00BE3A",
      "#439AE5",
      "#5400CE",
      "#5E0096",
      "#EF0067"
    ]
  }
};

export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
