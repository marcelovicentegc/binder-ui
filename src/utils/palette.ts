export const palette = [
  // Gray scale
  "#E9E9E9",
  "#C4C4C4",
  "#999999",
  "#4C4C4C",
  "#262626",
  // Red scale
  "#D24654",
  "#BB3B46",
  "#A52F38",
  "#831D23",
  // Orange scale
  "#EBA248",
  "#E2873C",
  "#D86C30",
  "#C9441E",
  "#C43618",
  // Yellow scale
  "#F4DA4E",
  "#EFC841",
  "#EAB634",
  "#DA7F0D",
  // Green scale
  "#5DC957",
  "#4DAE48",
  "#3E933A",
  "#276B24",
  "#0F420E",
  // Blue scale
  "#6ACEDF",
  "#59B5CF",
  "#479DBF",
  "#3584AF",
  "#12538F"
];
