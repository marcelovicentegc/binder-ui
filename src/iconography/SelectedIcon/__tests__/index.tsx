import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { SelectedIcon } from "..";
import { theme } from "../../../utils/Theme";

describe("<SelectedIcon /> test case", () => {
  test("test id is in the document", () => {
    const { getByTestId } = render(<SelectedIcon />);
    expect(getByTestId("selectedIcon")).toBeInTheDocument();
  });

  test("renders dimensions passed via props", () => {
    const { getByTestId } = render(<SelectedIcon dimension={40} />);
    expect(getByTestId("selectedIcon")).toMatchSnapshot();
  });
});
