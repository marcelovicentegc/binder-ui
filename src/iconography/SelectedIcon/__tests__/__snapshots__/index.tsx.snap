// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedIcon /> test case renders dimensions passed via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40px;
  height: 40px;
  background: white;
  box-shadow: 0px 0px 6px rgba(0,0,0,0.25);
  border-radius: 50%;
}

.c0 > svg {
  width: 20px;
  height: auto;
}

<div
  class="c0"
  data-testid="selectedIcon"
>
  <svg
    data-testid="arrowIcon"
    version="1.1"
    viewBox="0 0 451.846 451.847"
    x="0px"
    y="0px"
  >
    <g>
      <path
        d="M345.441,248.292L151.154,442.573c-12.359,12.365-32.397,12.365-44.75,0c-12.354-12.354-12.354-32.391,0-44.744   L278.318,225.92L106.409,54.017c-12.354-12.359-12.354-32.394,0-44.748c12.354-12.359,32.391-12.359,44.75,0l194.287,194.284   c6.177,6.18,9.262,14.271,9.262,22.366C354.708,234.018,351.617,242.115,345.441,248.292z"
        fill="#33333"
      />
    </g>
  </svg>
</div>
`;
